library/psa_crypto.c

```diff
@@ -3470,7 +3470,9 @@ psa_status_t psa_sign_hash_complete(
                                                    signature_length);
 
     /* Update ops count with work done. */
-    operation->num_ops = psa_driver_wrapper_sign_hash_get_num_ops(operation);
+    if (status == PSA_SUCCESS) {
```

Comment:
See below, we should call `psa_driver_wrapper_sign_hash_get_num_ops()` anyway and only overwrite status on failure.